frontend/src/views/Results.tsx

```diff
@@ -1,10 +1,99 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import styled from 'styled-components';
+import { useLocation, useHistory } from 'react-router-dom';
+import { Message } from 'stompjs';
+import { LargeText, Text } from '../components/core/Text';
+import { Game, Player, playAgain } from '../api/Game';
+import { checkLocationState } from '../util/Utility';
+import { errorHandler } from '../api/Error';
+import PlayerResultsCard from '../components/card/PlayerResultsCard';
+import { PrimaryButton } from '../components/core/Button';
+import { Room } from '../api/Room';
+import ErrorMessage from '../components/core/Error';
+import Loading from '../components/core/Loading';
+import { disconnect, routes, subscribe } from '../api/Socket';
+import { User } from '../api/User';
+
+const Content = styled.div`
+  padding: 0 20%;
+`;
+
+type LocationState = {
+  game: Game,
+  currentUser: User,
+};
 
 function GameResultsPage() {
+  const history = useHistory();
+  const location = useLocation<LocationState>();
+
+  const [error, setError] = useState('');
+  const [loading, setLoading] = useState(false);
+
+  const [players, setPlayers] = useState<Player[]>();
+  const [currentUser, setCurrentUser] = useState<User | null>(null);
+  const [room, setRoom] = useState<Room | null>(null);
+
+  useEffect(() => {
+    if (checkLocationState(location, 'game', 'currentUser')) {
+      setPlayers(location.state.game.players);
+      setRoom(location.state.game.room);
+      setCurrentUser(location.state.currentUser);
+
+      const subscribeCallback = (result: Message) => {
+        const updatedGame: Game = JSON.parse(result.body);
+
+        // Disconnect users from socket and then redirect them to the lobby page
+        if (updatedGame.playAgain) {
+          disconnect()
+            .then(() => {
+              history.replace(`/game/lobby?room=${updatedGame.room.roomId}`, {
+                user: location.state.currentUser,
+                roomId: updatedGame.room.roomId,
+              });
+            })
+            .catch((err) => setError(err.message));
+        }
+      };
+
+      subscribe(routes(location.state.game.room.roomId).subscribe_game, subscribeCallback)
+        .catch((err) => setError(err.message));
```

Comment:
Before subscribing the users to the "subscribe game" route again, unsubscribe them from the current "subscribe game" route, and from the "notifications" route. This can be done by passing the subscription through to the `Results` page in location / history, then calling `unsubscribe()` on that object.

Reply:
Sounds good - instead of passing those over location though, I just unsubscribed them on the game page before redirecting (I also changed the boolean `xxxSubscribed` to subscription objects using a null check to determine the same result)